chore(dev): Add wrappers for check/generate scripts to `vdev`

bruceg opened the pull request:
This adds wrappers for _all_ of the `check-*` and `generate-*` scripts into `vdev`.

It also adds some pretty heavy macro helpers to avoid repeating the module or subcommand names three or four times each. The macros are kinda messy, but the resulting usages are as clean as I could make them.

<!--
**Your PR title must conform to the conventional commit spec!**

  <type>(<scope>)!: <description>

  * `type` = chore, enhancement, feat, fix, docs
  * `!` = OPTIONAL: signals a breaking change
  * `scope` = Optional when `type` is "chore" or "docs", available scopes https://github.com/vectordotdev/vector/blob/master/.github/semantic.yml#L20
  * `description` = short description of the change

Examples:

  * enhancement(file source): Add `sort` option to sort discovered files
  * feat(new source): Initial `statsd` source
  * fix(file source): Fix a bug discovering new files
  * chore(external docs): Clarify `batch_size` option
-->

spencergilbert commented on scripts/fmt.sh:
```diff
@@ -10,5 +10,5 @@ set -euo pipefail
 cd "$(dirname "${BASH_SOURCE[0]}")/.."
 set -x
 
-scripts/check-style.sh --fix
+cargo vdev check style --fix
 cargo fmt
```
This seems like another script that could be completely removed?

spencergilbert commented on scripts/release-prepare.sh:
```diff
@@ -13,5 +13,5 @@ set -eu
 cd "$(dirname "${BASH_SOURCE[0]}")/.."
 set -x
 
-scripts/generate-manifests.sh
-scripts/generate-release-cue.rb
+cargo vdev generate manifests
+cargo vdev generate release-cue
```
This seems like another script that could be completely removed?

spencergilbert commented on vdev/src/commands/check/fmt.rs:
```diff
@@ -0,0 +1,14 @@
+use anyhow::Result;
+
+use crate::app;
+
+/// Check that all files are formatted properly
+#[derive(clap::Args, Debug)]
+#[command()]
+pub struct Cli {}
+
+impl Cli {
+    pub fn exec(self) -> Result<()> {
+        app::exec_app_path("cargo", ["fmt", "--", "--check"])
+    }
+}
```
Does it actually make sense to have this function? In the `script/fmt.sh` we directly call `cargo fmt` to do the formatting, why don't we do the same for checks?

spencergilbert commented on vdev/src/commands/release/prepare.rs:
```diff
@@ -0,0 +1,17 @@
+
+impl Cli {
+    pub fn exec(self) -> Result<()> {
+        // Dummy arguments to satisfy `impl` trait bound
```
That's a little awkward, how much effort would it be to allow that to be optional?

bruceg commented:
Okay, I think the churn on this should be done. Moving on now…

spencergilbert reviewed (commented):
I'll play with this some today and make sure everything works in the daily workflow.

I still feel odd about creating wrappers around `cargo fmt` and `cargo fmt --check` but admittedly it's probably easier to just say "`vdev` has all needed commands, regardless of simplicity" rather than nitpicking on what qualifies as "simple enough".

spencergilbert commented:
🤔 
```shell
vector on  bruceg/vdev-wrappers is 📦 v0.27.0 via 🦀 v1.66.0
❯ cargo vdev -vvv run /tmp/vector.toml
error: Found argument '--no-default-features' which wasn't expected, or isn't valid in this context

Usage: cargo [+toolchain] [OPTIONS] [COMMAND]

For more information try '--help'
Error: Command failed, exit code exit status: 1
```

1. I'm not sure why that command is wrong..
2. A nice UX improvement could be emitting the exec'd command when verbosity is increased

bruceg commented:
> I'll play with this some today and make sure everything works in the daily workflow.

Thanks.

> I still feel odd about creating wrappers around `cargo fmt` and `cargo fmt --check` but admittedly it's probably easier to just say "`vdev` has all needed commands, regardless of simplicity" rather than nitpicking on what qualifies as "simple enough".

If it was just `cargo fmt` I'd agree. In the case of `vdev fmt`, it also runs `scripts/check-style.sh`.

>     1. I'm not sure why that command is wrong..

Missing `"run"`, fixed.

>     2. A nice UX improvement could be emitting the exec'd command when verbosity is increased

Agreed. I was going to do that as a separate PR, but I've added it now.

spencergilbert commented:
🎨 

```shell
❯ cargo vdev -vvv run /tmp/vector.toml
Executing: "cargo" "run" "--no-default-features" "--features" "sinks-socket,sources-stdin" "--" "--config" "/tmp/vector.toml"
```

spencergilbert reviewed (approved):
Worked with everything I tried, didn't fully exercise the release commands though - perhaps @jszwedko should weigh in before we remove the OG scripts?
